### DISCUSSION TITLE
Lesson 9: testCantEnterWhenRaffleIsCalculating

### DISCUSSION THREAD

**Discussion by neo-t-maredi** [Cyfrin/foundry-full-course-cu#1758] [Q&A]
Hey There. 

Things are seemingly going well. Unfortunately on lesson 9, I'm receiving a (2314) error as follows however: 

![Screenshot 2024-05-28 094301](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/f72695da-b587-4809-8cbb-76996ceb74c3)
![Screenshot 2024-05-28 094336](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/2744414b-8d74-4d71-b476-5260f8c47ccb)
![Screenshot 2024-05-28 094400](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/4c391fd4-4e49-4424-963d-3be228e14f65)
![Screenshot 2024-05-28 094422](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/e8ea59c1-443e-4c33-870d-661bb72daceb)

Any help would be greatly appreciated. Amazing course!

**Comment by EngrPips** [comment]
Hello @neo-t-maredi, You have the `addConsumerUsingConfig(raffle)` in the contract global space instead of inside the `run` function. Put it inside the `run` function and let us know if that resolves the issue.

**Reply by neo-t-maredi** [reply]
No it's throwing an error (7920)

**Reply by neo-t-maredi** [reply]
![Screenshot 2024-05-30 091721](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/c7490586-e415-48fd-8c69-9ac714f1ebd5)
![Screenshot 2024-05-30 091738](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/97213a1b-5ab7-494b-a0c7-b39ac37a94ce)

**Reply by neo-t-maredi** [reply]
I also moved the addConsumerUsingConfig(), as suggested and it's still throwing an error. Not sure big man (or person, don't want to overassume) :/
![Screenshot 2024-05-30 092230](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/bd48cb7d-0674-4659-8d6e-42c2950866d4)

**Reply by urosognjenovic** [reply]
Can you share your source code with us to help you debug the contract?

**Reply by EngrPips** [reply]
Hello @neo-t-maredi, Please show your `Interaction.s.sol` file
